ophyd/device.py

```diff
@@ -424,8 +424,19 @@ def stage(self):
         logger.debug("Staging %s", self.name)
         self._staged = Staged.partially
 
+        # Resolve any stage_sigs keys given as strings: 'a.b' -> self.a.b
+        stage_sigs = OrderedDict()
+        for k, v in self.stage_sigs.items():
+            if isinstance(k, str):
+                attr = self
```

Comment:
`Device.__getattr__` takes care of this stuff already - pass it the dotted name directly